src/Features/CSharp/Portable/SignatureHelp/InitializerExpressionSignatureHelpProvider.cs

```diff
@@ -0,0 +1,124 @@
+               token.Parent is InitializerExpressionSyntax;
+
+        private static bool IsInitializerExpressionToken(InitializerExpressionSyntax expression, SyntaxToken token)
+            => expression.Span.Contains(token.SpanStart) && token != expression.CloseBraceToken;
+
+        protected override async Task<SignatureHelpItems> GetItemsWorkerAsync(Document document, int position, SignatureHelpTriggerInfo triggerInfo, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (!TryGetInitializerExpression(root, position, document.GetLanguageService<ISyntaxFactsService>(), triggerInfo.TriggerReason, cancellationToken, out var initializerExpression))
+            {
+                return null;
+            }
+
+            var semanticModel = await document.GetSemanticModelForNodeAsync(initializerExpression, cancellationToken).ConfigureAwait(false);
+            var compilation = semanticModel.Compilation;
+            var ienumerableType = compilation.GetTypeByMetadataName(typeof(IEnumerable).FullName);
+            if (ienumerableType == null)
+            {
+                return null;
+            }
+
+            // get the regular signature help items
+            var parentOperation = semanticModel.GetOperation(initializerExpression.Parent, cancellationToken) as IObjectOrCollectionInitializerOperation;
+            var parentType = parentOperation?.Type;
+            if (parentType == null)
+            {
+                return null;
+            }
+
+            if (!parentType.AllInterfaces.Contains(ienumerableType))
+            {
+                return null;
+            }
+
+            var addSymbols = semanticModel.LookupSymbols(
+                position, parentType, WellKnownMemberNames.CollectionInitializerAddMethodName, includeReducedExtensionMethods: true);
+
+            // We want all the accessible '.Add' methods that take at least two arguments. For
+            // example, say there is:
+            //
+            //      new JObject { { $$ } }
+            //
+            // Technically, the user could be calling the `.Add(object)` overload in this case.
+            // However, normally in that case, they would just supply the value directly like so:
+            //
+            //      new JObject { new JProperty(...), new JProperty(...) }
+            //
+            // So, it's a strong signal when they're inside another `{ $$ }` that they want to
+            // call the .Add methods that take multiple args, like so:
+            //
+            //      new JObject { { propName, propValue }, { propName, propValue } }
+
+            var symbolDisplayService = document.GetLanguageService<ISymbolDisplayService>();
+            var addMethods = addSymbols.OfType<IMethodSymbol>()
+                                       .Where(m => m.Parameters.Length >= 2)
```

Comment:
>Where [](start = 40, length = 5)

nit: WhereAsArray

Reply:
Not actually helpful here.  It would just realize the intermediary collection unnecessarily.  note that for sighelp we're also only generally talking about a small handful of overloads.

Reply:
unfortunately not applicable.  taht is used when you have an ImmutableArray and you want to .Where off of it into a new immutbale array.  Here we have an ienumerable.  so we do the .Where, and then the ToImmutable.